src/java/org/apache/cassandra/index/sai/disk/PostingListRangeIterator.java

```diff
@@ -82,6 +87,8 @@ public PostingListRangeIterator(IndexContext indexContext,
         this.postingList = searcherContext.postingList;
         this.searcherContext = searcherContext;
         this.queryContext = this.searcherContext.context;
+        // Get and store reference to scores map if it exists.
+        this.ssTableRowIdToScoreMap = this.queryContext.getScoreCacheForSSTable(sstableId);
```

Comment:
Would it make more sense to put this logic in the `IndexSearcherContext`? This would leave the `PostingListRangeIterator` pretty much untouched. We could go as far as putting the `PrimaryKeyMap` in the context as well and reducing the call in `computeNext` to `return searcherContext.toPrimaryKey(rowId)` with all the score recording done in the context.

Reply:
I wasn't able to abstract it away as much as you mentioned, but I think I found a good middle way solution. Let me know what you think.